litestar/channels/memory.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+import asyncio
+import contextlib
+from asyncio import Queue
+from collections import defaultdict, deque
+from typing import Any, AsyncGenerator, Iterable
+
+from litestar.channels.base import ChannelsBackend
+
+
+class MemoryChannelsBackend(ChannelsBackend):
+    """An in-memory channels backend"""
+
+    def __init__(self, history: int = 0) -> None:
+        self._max_history_length = history
+        self._channels: set[str] = set()
+        self._queue: Queue[tuple[str, bytes]] | None = None
+        self._history: defaultdict[str, deque[bytes]] = defaultdict(lambda: deque(maxlen=self._max_history_length))
+
+    async def on_startup(self) -> None:
+        self._queue = Queue()
+
+    async def on_shutdown(self) -> None:
+        self._queue = None
+
+    async def publish(self, data: bytes, channels: Iterable[str]) -> None:
+        """Publish ``data`` to ``channels``. If a channel has not yet been subscribed to,
+        this will be a no-op.
+
+        Args:
+            data: Data to publish
+            channels: Channels to publish to
+
+        Returns:
+            None
+
+        Raises:
+            RuntimeError: If ``on_startup`` has not been called yet
+        """
+        if not self._queue:
+            raise RuntimeError("Backend not yet initialized. Did you forget to call on_startup?")
```

Comment:
`ImproperlyConfiguredException`

Reply:
Not really. It's not a configuration error. We follow the same pattern elsewhere.